altair/utils/data.py

```diff
@@ -148,19 +150,28 @@ def to_values(data):
     elif isinstance(data, pd.DataFrame):
         data = sanitize_dataframe(data)
         return {"values": data.to_dict(orient="records")}
-    elif isinstance(data, dict):
-        if "values" not in data:
-            raise KeyError("values expected in data dict, but not present.")
+    elif isinstance(data, pd.Series):
+        data = sanitize_dataframe(data.reset_index())
+        return {"values": data.to_dict(orient="records")}
+    elif isinstance(data, dict) and "values" in data:
         return data
+    elif isinstance(data, dict):
+        data = pd.DataFrame(data)
+        data = sanitize_dataframe(data)
+        return {"values": data.to_dict(orient="records")}
+    elif isinstance(data, list):
+        data = pd.DataFrame(data, columns=[f"x{i}" for i in range(len(data))])
```

Comment:
I think I'd prefer not to support generic lists like this. In particular, given how central column names are to Altair chart encodings, it seems unhelpful to generate generic ones in a way that is opaque to the user.